src/services/formatting/formatting.ts

```diff
@@ -360,7 +360,9 @@ namespace ts.formatting {
             range: TextRange, 
             inheritedIndentation: number): number {
             
-            if (rangeOverlapsWithStartEnd(range, startPos, endPos)) {
+            if (rangeOverlapsWithStartEnd(range, startPos, endPos) ||
+                rangeContainsStartEnd(range, startPos, endPos) /* Not to miss zero-range nodes e.g. JsxText */) {

```

Comment:
@mhegazy I've found that this code blocks autoformatting:

```typescript
let x = <div>
<span>
</span>
</div>
```

`<span>` won't be indented because by some reason a zero-length JsxText node before JsxElement node incorrectly sets indentation value to zero. An additional `rangeContainsStartEnd` check prevents this bug.